Gruntfile.js

```diff
@@ -40,6 +40,35 @@ var getFromExec = function (command, options) {
     return result;
 };
 
+/**
+ * Adds '.min' convention in front of the first period of a filename string
+ * Example results:
+ * infusion-all.js -> infusion-all.min.js
+ * infusion-all.js.map -> infusion-all.min.js.map
+ * @param {String} fileName - filename string to add '.min' to
+ * @returns the modified filename string
+ */
+var addMin = function (fileName) {
+    var segs = fileName.split(".");
+    var min = "min";
+
+    if (segs[0] && segs.indexOf(min) < 0 ) {
+        segs.splice(1, 0, min);
+    }
+    return segs.join(".");
+};
+
+/**
+ * Rename function for grunt file tasks for  adding ".min" convention
+ * to filename string; won't do anything to strings that already
+ * include ".min"
+ * @param {String} dest - supplied by Grunt task, see http://gruntjs.com/configuring-tasks#the-rename-property
+ * @param {String} src - supplied by Grunt task, see http://gruntjs.com/configuring-tasks#the-rename-property
+*/
+var addMinifyToFilename = function (dest, src) {
+    return dest + addMin(src);
```

Comment:
Tested on Windows 10 and works (dest & src are supplied by the Grunt task, so presumably take advantage of Grunt's cross-platform capability).